server/src/main/kotlin/org/javacs/kt/KotlinLanguageServer.kt

```diff
@@ -72,6 +72,14 @@ class KotlinLanguageServer(
     fun getProtocolExtensionService(): KotlinProtocolExtensions = protocolExtensions
 
     override fun initialize(params: InitializeParams): CompletableFuture<InitializeResult> = async.compute {
+        // Parse initialization options
+        val options = getInitializationOptions(params)
+        databaseService.setup(options?.storagePath)
+
+        if(options?.additionalSourceExclusions != null) {
+            sourceFiles.updateAdditionalExclusions(options.additionalSourceExclusions.toMutableList())
```

Comment:
Can we refactor this to use immutable lists?

Mutable structures make it harder to reason about where updates are happening and passing them around make it easier for things to get out-of-sync, therefore I usually prefer the functional style.